src/kibana/plugins/visualize/editor/editor.js

```diff
@@ -125,6 +125,46 @@ define(function (require) {
 
       editableVis.listeners.click = vis.listeners.click = filterBarClickHandler($state);
       editableVis.listeners.brush = vis.listeners.brush = brushEvent;
+      editableVis.listeners.mapZoomEnd = vis.listeners.mapZoomEnd = function (event) {
+        if (!vis.params.autoPrecision) return;
+
+        var geoHash = _.find(vis.aggs, function (agg) {
+          return agg.type.name === 'geohash_grid';
+        });
+
+        geoHash.params.precision = autoPrecision(event.zoom, config.get('visualization:tileMap:maxPrecision'));
+        $scope.fetch();
+      };
+
+      function autoPrecision(zoom, limit) {
+        var precision;
+
+        // zoomPrecision keys are map zoom, values are geohash precision
+        // default max precision is 7, configurable up to 12
+        // limit (max precision) is the max precision value returned
+        var zoomPrecision = {
+          1: 1,
```

Comment:
I agree, will set min auto precision to 2 and make a configurable base precision.